src/infinity/libraries/TokenSwap/LibUniswapV2Swap.sol

```diff
@@ -16,15 +17,29 @@ library LibUniswapV2Swap {
         uint amountIn,
         uint amountOut
     ) internal returns (uint) {
+        AppStorage storage s = LibAppStorage.diamondStorage();
         LibApprove.approveToken(IERC20(path[0]), router, amountIn);
 
-        uint[] memory amountsOut = IUniswapV2Router02(router).swapTokensForExactTokens(
-            amountOut,
-            amountIn,
-            path,
-            address(this),
-            block.timestamp
-        );
+        uint[] memory amountsOut;
+
+        // TODO [ REVIEW & TEST]
+        if (s.swap[path[1]].transferFee) {
+            amountsOut = IUniswapV2Router02(router).swapTokensForExactTokens(
+                amountOut,
+                amountIn,
+                path,
+                address(this),
+                block.timestamp
+            );
+        } else {
+            IUniswapV2Router02(router).swapTokensForExactTokens(
+                amountOut,
```

Comment:
should the else condition call a different function?

Reply:
That's sorted now. 
Can you please check?